### PR TITLE
New tip for demand graph in EIA (US Energy) example

### PR DESCRIPTION
**Pull request by pettiross**
Replaces default tip for the three-line graph with an hourly summary so it's easier to read. Also changed to zip archive data loader to create 2 different levels of detail from a single web API call.

Apologies in advance for the rename of us-demand.csv.js to us-demand.zip.js not showing as a diff. I tried using both `git mv` on the CLI and the GitHub UI to rename the file (see the 28 Feb commits), but if there was a way to make it work, I did not figure it out. 🤷

image

Thank you to both @Fil and @allisonhorst for their patient pre-review of this change. ❤️

### CODE REVIEW

**Comment by pettiross** [comment]
Thanks for the comments, @Fil. As you suggested, I changed the data loader back to a single CSV and moved the rollup into the chart code (which also cleared up the need to deal with the rename). I also updated the date in the tooltip to be like "10 Mar" (which is how I tend to write dates personally anyway...).

File: examples/eia/docs/components/charts.js [hunk 1]
```diff
@@ -20,28 +26,67 @@ export function top5BalancingAuthoritiesChart(width, height, top5Demand, maxDema
         title: ({ name, value }) => `name: ${name}\ndemand: ${value / 1000} GWh`
       })
     ]
-  });
+  })
```
Comment (Fil) [inline examples/eia/docs/components/charts.js:29]:
```suggestion
  });
```

File: examples/eia/docs/components/charts.js [hunk 2]
```diff
@@ -20,28 +26,67 @@ export function top5BalancingAuthoritiesChart(width, height, top5Demand, maxDema
...
 // US electricity demand, generation and forecasting chart
-export function usGenDemandForecastChart(width, height, usDemandGenForecast, currentHour) {
+export function usGenDemandForecastChart(width, height, data, currentHour) {
+  
+  // Roll up each hour's values into a single row for a cohesive tip
+  const rolledToHour = data.reduce((map, d) => {
+    let value = map.get(d.date.getTime()) ?? { date: d.date, demandForecast: null, demandActual: null, netGeneration: null };
+    value[d.name] = d.value;
+    return map.set(d.date.getTime() , value);
+  }, new Map())
```
Comment (Fil) [inline examples/eia/docs/components/charts.js:40]:
```suggestion
  }, new Map());
```

File: examples/eia/docs/components/charts.js [hunk 3]
```diff
@@ -20,28 +26,67 @@ export function top5BalancingAuthoritiesChart(width, height, top5Demand, maxDema
...
+
+  const rolledUpSparse = [];
+  rolledToHour.forEach((d) => { rolledUpSparse.push(d) });
```
Comment (Fil) [inline examples/eia/docs/components/charts.js:43]:
```suggestion
  for (const d of rolledToHour) rolledUpSparse.push(d);
```

File: examples/eia/docs/components/charts.js [hunk 4]
```diff
@@ -20,28 +26,67 @@ export function top5BalancingAuthoritiesChart(width, height, top5Demand, maxDema
...
-    y: {label: null},
-    x: {type: "time", tickSize: 0, tickPadding: 3},
+    y: {tickSize: 0, label: null},
+    x: {type: "time", tickSize: 0, tickPadding: 3, label: null},
     color: {
       legend: true,
-      domain: ["Day-ahead demand forecast", "Demand", "Net generation"],
-      range: ["#6cc5b0", "#ff8ab7", "#a463f2"]
+      domain: ["demandActual", "demandForecast", "netGeneration"],
+      tickFormat: d => friendlyTypeName[d],
```
Comment (Fil) [inline examples/eia/docs/components/charts.js:54]:
```suggestion
      tickFormat: (d) => friendlyTypeName[d],
```

File: examples/eia/docs/components/charts.js [hunk 5]
```diff
@@ -20,28 +26,67 @@ export function top5BalancingAuthoritiesChart(width, height, top5Demand, maxDema
...
+            x: false
+          },
+          fontSize: 12,
+          anchor: "bottom",
+          frameAnchor: "top"
+        }
+      }))
     ]
-  });
+  })
```
Comment (Fil) [inline examples/eia/docs/components/charts.js:89]:
```suggestion
  });
```

File: examples/eia/docs/components/charts.js [hunk 6]
```diff
@@ -65,5 +110,5 @@ export function countryInterchangeChart(width, height, usDemandGenForecast, coun
       }),
       Plot.ruleY([0], {strokeOpacity: 0.3})
     ]
-  });
+  })
```
Comment (Fil) [inline examples/eia/docs/components/charts.js:113]:
```suggestion
  });
```

**Review by Fil — APPROVED** [review]

**Review by Fil — CHANGES_REQUESTED** [review]
one sec, I need to test something

**Review by Fil — APPROVED** [review]

**Comment by Fil** [comment]
it is now deployed https://observablehq.com/framework/examples/eia/

**Comment by pettiross** [comment]
Thank you for the further edits, @Fil. I'm learning by reviewing what you did.

**Pull request merged**